src/Symfony/Component/AccessToken/AccessTokenFetcher.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AccessToken;
+
+/**
+ * @author Pierre Rineau <pierre.rineau@processus.org>
+ */
+class AccessTokenFetcher
+{
+    public function __construct(
+        private readonly AccessTokenManagerInterface $accessTokenManager,
+        private readonly CredentialsInterface $credentials,
+    ) {
+    }
+
+    /**
+     * Create instance using given URI.
+     */
+    public static function createWithUri(AccessTokenManagerInterface $accessTokenManager, #[\SensitiveParameter] string $uri): self
+    {
+        return new self($accessTokenManager, $accessTokenManager->createCredentials($uri));
+    }
+
+    /**
+     * Create credentials interface with given URI.
+     */
+    public static function createCredentialsWithUri(AccessTokenManagerInterface $accessTokenManager, #[\SensitiveParameter] string $uri): CredentialsInterface
```

Comment:
That's actually a good idea, indeed. But you have to take into account that it also injects the `CredentialsInterface` instance as a service, you can either inject the fetcher, or inject the credentials directly, it's up to the user.

Maybe it could be reversed, inject the fetch with the credential as string, then use a factory method for the credentials interface ?

Reply:
not sure i understand: to be able to call this method, i will need the AccessTokenManagerInterface instance. and when i have that instance, i can directly call the method on that instance instead.

is the idea that the AccessTokenFetcher can be overwritten to do additional things? that would be a bit fragile imho.

Reply:
I should have documented better, that's just a factory for the container, but it needs to be publicly available. If you have a suggestion to make this clearer, I'd be happy to write it better.

Reply:
May be I should simply document it and mark it internal ?

Reply:
does the container need to use this and pass the AccessTokenManagerInterface, but can not call the method on the manager directly?

Reply:
We cross-posted, that exactly what I did in the end, and it's much more readable indeed. It also removes a lot of boilerplate useless code and some unit tests that weren't necessary. Thanks !